### ISSUE TITLE
java.lang.Exception: Invalid ScalaCheck args

### ISSUE THREAD

**Issue by aloiscochard** [rickynils/scalacheck#173]
Hi,

When upgrading to 1.12.3 from 1.12.1 I now have the following exception `(web/test:executeTests) java.lang.Exception: Invalid ScalaCheck args`.

Thanks

**Comment by rickynils** [comment]
@aloiscochard See https://github.com/rickynils/scalacheck/issues/165#issuecomment-104588601 . Are you using ScalaJS? In earlier versions of ScalaCheck, the test arguments was not passed on for ScalaJS runs. You can try using 1.12.4-SNAPSHOT to get a bit mor info about what parameters might cause the exception.

**Comment by aloiscochard** [comment]
@rickynils okay, thanks to 1.12.4 I know have the following information:

`java.lang.Exception: Invalid ScalaCheck args: List(-oF)`

**Comment by aloiscochard** [comment]
Okay, it's an integration issue with scalatest, probably somethnig changed in SBT. I fixed this by replacing:
`testOptions in Test += Tests.Argument("-oF")`

with:
`testOptions in Test += Tests.Argument(TestFrameworks.ScalaTest, "-oF")`

http://www.scalatest.org/user_guide/using_scalatest_with_sbt

**Comment by limansky** [comment]
I've observed that running individual test is works fine with ScalaCheck 1.12.2, but brocken with 1.12.4. I'm doing it in this way:

testOnly MyClassTest -- -z "my test name"

I propose to reopen this issue.

**Comment by rickynils** [comment]
@limansky `-z "my test name"` is not a valid ScalaCheck option. Can you tell us a bit more what you are trying to achieve and what error you get?

**Comment by limansky** [comment]
@rickynils Let me show you an example. Assume I'd like ot run only one test from big test suite for debuging purpose. Here is a real example from https://github.com/limansky/mongoquery with ScalaCheck 1.12.2:
```
> core/testOnly com.github.limansky.mongoquery.core.bsonparser.ParserTest -- -z "parse int values"
[info] ParserTest:
[info] BSON Parser
[info] - should parse int values
[info] ScalaTest
[info] Run completed in 843 milliseconds.
[info] Total number of tests run: 1
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 1, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[info] Passed: Total 1, Failed 0, Errors 0, Passed 1
[success] Total time: 2 s, completed 06.07.2015 17:45:12
```
Now, the same with the ScalaCheck 1.12.4:
```
> core/testOnly com.github.limansky.mongoquery.core.bsonparser.ParserTest -- -z "parse int values"
[trace] Stack trace suppressed: run last core/test:testOnly for the full output.
[error] (core/test:testOnly) java.lang.Exception: Invalid ScalaCheck args: List(-z, parse int values)
[error] Total time: 0 s, completed 06.07.2015 17:52:18
```

**Comment by rickynils** [comment]
@limansky `java.lang.Exception: Invalid ScalaCheck args: List(-z, parse int values)` tells you why ScalaCheck fails. `-z "parse int values"` is not an option that ScalaCheck recognizes. For some reason it is handed to ScalaCheck by scalatest/sbt, and that is the problem in the first place. As you can see above, @aloiscochard has a workaround for it.

**Comment by limansky** [comment]
Well, workaround is ok for permanent options, but it is not an option for temporary settings, like test name for running. I try to investigate it if I'll have free time.

**Comment by jroper** [comment]
@rickynils it appears that ScalaCheck is assuming that it is the only test runner, and throwing exceptions for arguments it doesn't understand. This is simply not true. Prior to upgrading ScalaCheck from 1.2.12 to 1.2.14, I could run individual specs2 examples in the sbt console by doing this:

    test-only com.example.MySpec -- -ex "this should work"

But now, I get this error:

    java.lang.Exception: Invalid ScalaCheck args: List(-ex, this should work)

So it appears that ScalaCheck should be ignoring arguments that it doesn't understand, not failing, because failing interferes with the normal operation of other sbt test runners.

**Comment by rickynils** [comment]
@jroper I see. I have to look into it. It could be that ScalaCheck picks up tests that it shouldn't. Or that sbt hands ScalaCheck options that it shouldn't.

**Comment by xuwei-k** [comment]
I completely agree with jroper.

**Comment by rickynils** [comment]
I've realized sbt creates Runners for all test frameworks even though no tests are found, so ScalaCheck is doing the wrong thing here.

I'll fix it, but the question is how to handle faulty arguments. I'll probably ignore them and just log a warning message (and only do this for actual ScalaCheck tests)

**Comment by jedws** [comment]
hmm, pretty annoying for us too, makes the upgrade impossible

**Comment by rickynils** [comment]
I've resolved this both on the `master` branch (version `1.13.0-SNAPSHOT`), and back-ported it to `1.12.x`, so it will be released in `1.12.5` too.

**Comment by andriichuk** [comment]
Hi, @rickynils. 

Unfortunately I experience the `Invalid ScalaCheck args` error even on `1.13.0`. 

I use the following command: 
```sbt "test-only com.xxx.DosageLettersTest -- -DmongoHost=localhost -DpgHost=localhost"```

Do I do something wrong?
